src/Model/Repository/BaseRepository.php

```diff
@@ -0,0 +1,116 @@
+<?php
+
+
+namespace Friendica\Model\Repository;
+
+use Friendica\Database\Database;
+use Friendica\Model\Entity\BaseEntity;
+use Friendica\Model\Entity\IStorable;
+
+abstract class BaseRepository
+{
+	/** @var string */
+	protected static $table;
+	/** @var BaseEntity */
+	protected static $entity;
+
+	/** @var Database */
+	protected $dba;
+
+	/**
+	 * BaseRepository constructor.
+	 *
+	 * @param Database $dba
+	 */
+	public function __construct(Database $dba)
+	{
+		$this->dba = $dba;
+	}
+
+	/**
+	 * @param array $data
+	 *
+	 * @return mixed
+	 */
+	abstract protected function createEntity(array $data);
+
+	/**
+	 * @param array $conditions
+	 *
+	 * @return mixed
+	 *
+	 * @throws \Exception
+	 */
+	public function select(array $conditions)
+	{
+		$prototype = null;
+
+		$selected = $this->dba->delete(static::$table, [], $conditions);
```

Comment:
idea: we can use defined field-lists dependending on different usages per repository, like
`selectIntrosWithContactData()`
`selectIntrosOnly()`